src/compiler/checker.ts

```diff
@@ -5665,6 +5665,12 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             if (getSymbolIfSameReference(exported, symbol)) {
                 return exported;
             }
+            if (symbol.flags & SymbolFlags.TypeAlias && exported.declarations?.find(isTypeAlias)) {
+                const aliasSymbol = getDeclaredTypeOfTypeAlias(exported).aliasSymbol;
+                if (aliasSymbol && getSymbolIfSameReference(aliasSymbol, symbol)) {

```

Comment:
Shouldn't this be a change in the logic of `getSymbolIfSameReference` instead? To make a type alias that points directly at another type count as a "same reference" of that type? It's been reused in one or two other places, but it basically only exists to answer "should we consider these two symbols as ultimately referring to the same thing", and mostly just for the usages here, in this function. In particular, only doing this check here means 1. the fastpath lookup will fail for type aliases that pass this check, so they always hit the slow path and 2. an `export =`'d type alias will *never* be considered equal to the target symbol.